### PR TITLE
Add an option to ignore the Reacli configuration file

### LINKED ISSUE #45: Add an option to ignore the configuration file
**Is your feature request related to a problem? Please describe.**
Today, when using the CLI (or in the near future when using the interactive CLI, see #43), Reacli detects if a `.reacli` file is present.

It would be nice to add an option like `--ignore-config-file` to prevent the CLI to take the `.reacli` file into account and only accept the options passed at runtime to the CLI.

### PR DESCRIPTION
**Pull request by gdarchen**
# Fixes #45 

**Summary:**

I added a `-i` or `--ignore-config-file` option. This argument can be used either in interactive CLI or directly in command line.

It enables to not take the optional `.reacli` file into account if this one is present. If the option is used but there is no `.reacli` file in the project, it doesn't do anything.

For example, on the following picture, I used the `-i` argument twice, the first time in an _inline_ mode, and the second time with the _interactive_ CLI.
image

As you can see, one blue log line tells the user that if a config file is there, it just will be ignored.

In my example, I had a `.reacli` file with the following content:
```json
{
  "flow": true
}
```
and the generated component only uses the `--redux` option which is given at runtime in the command line 😊 

**Changes:**

* Reformat code to place in a separate function the action of ignoring the configuration file if it exists
* Add the `-i` or `--ignore-config-file` argument
* Make the interactive CLI also accept not to use the `.reacli` file if the `-i` option is used
* Add the `.reacli` file to `.gitignore` to make sure that we don't push this file that we might have been created while testing our feature

**Dependencies:**

#43 

**Instructions:**

Create a `.reacli` configuration file and try using the CLI in both modes, with and without the `-i` option to see if it ignores the `.reacli` file correctly.

### CODE REVIEW

**Comment by tanohzana** [comment]
There appear to be some conflicts... Gotta fix them before merging... 🤗

**Review by tanohzana — CHANGES_REQUESTED** [review]
There appear to be some conflicts... Gotta fix them before merging... 🤗

**Comment by gdarchen** [comment]
> There appear to be some conflicts... Gotta fix them before merging... 🤗

It should be fixed with https://github.com/reacli/cli/pull/48/commits/c47e367350d80d213fa463071b08b4be7ec20922 😊

**Comment by tanohzana** [comment]
- [x] feature is working
 - [x] no breaking change was caused
 - [x] code is clean
 - [x] new tests pass

File: lib/core.js [hunk 1]
```diff
@@ -23,6 +23,16 @@ const loadOptionsInConfigFileIfExists = async (path, options = {}, componentName
 	return options
 }
 
+const loadReacliConfigurationFileIfNotIgnored = async (ignoreConfigFile, path, options, componentName = "") => {
```
Comment (tanohzana) [inline lib/core.js]: I think the function's name gives too much details about what it does. In my opinion, `loadReacliConfiguration` is enough and ignoreConfigFile could be added as an optional parameter at the end as such:

`loadReacliConfiguration = async (path, options, componentName = "", ignoreConfigFile)`

File: lib/core.js [hunk 2]
```diff
@@ -23,6 +23,16 @@ const loadOptionsInConfigFileIfExists = async (path, options = {}, componentName
...
+	if (ignoreConfigFile) {
+		console.log(chalk.black.bgBlue("Ignore the optional Reacli configuration file."))
+	} else {
+		options = await loadOptionsInConfigFileIfExists(path, options, componentName)
```
Comment (tanohzana) [inline lib/core.js]: Same here... In my opinion, `loadOptionsInConfigFile` is greatly enough ☺️

**Review by tanohzana — CHANGES_REQUESTED** [review]

File: lib/core.js (hunk 1 above)
Comment (gdarchen) [reply lib/core.js]: https://github.com/reacli/cli/pull/48/commits/b221ae0995396f63c2aca2af55e80e10ee2083e5

File: lib/core.js (hunk 2 above)
Comment (gdarchen) [reply lib/core.js]: https://github.com/reacli/cli/pull/48/commits/6834c0917e02506b0492d5f7a2bead3c168de52f

**Review by tanohzana — CHANGES_REQUESTED** [review]
Just a small conflict. Easy to fix 😉

**Comment by gdarchen** [comment]
> Just a small conflict. Easy to fix 😉

https://github.com/reacli/cli/pull/48/commits/a1c26f1b8cf47bee487d5b4212a9cd0275798f66

**Review by tanohzana — APPROVED** [review]

**Pull request merged**